lib/rules/entry-points.js

```diff
@@ -61,6 +61,28 @@ module.exports = {
           }
         }
 
+        if (returnArgument && returnArgument.type === 'Identifier') {
+          for (let i = 0; i < callbackBody.length; i++) {
+            const childNode = callbackBody[i];
+
+            if (childNode.type !== 'ExpressionStatement') {
+              continue;
+            }
+
+            const expressionLeft = childNode.expression.left;
+
+            if (
+              expressionLeft.object.name === returnArgument.name &&
+              scriptTypeMap[scriptType].entryPoints.includes(expressionLeft.property.name) &&
+              (expressionLeft.property.type === 'Identifier' ||
+              expressionLeft.property.type === 'FunctionExpression')
+            ) {
+              hasValidEntryPoint = true;
+              break;
+            }
+          }
```

Comment:
This could be simplified to a filter on `callbackBody`.
```suggestion
          const returnAssignments = callbackBody.filter(n =>
            n.type === 'ExpressionStatement' &&
            n.expression.left.type === 'MemberExpression' &&
            n.expression.left.object.name === returnArgument.name &&
            scriptTypeMap[scriptType].entryPoints.includes(n.expression.left.property.name)
          );

          if (returnAssignments.length > 0) {
            hasValidEntryPoint = true;
          }
```